modules/time-shortcode/load.php

```diff
@@ -10,114 +10,150 @@
+		add_filter( 'comment_text', array( 'o2_Time_Shortcode', 'do_comment_time_shortcode' ) );
 	}
-	$time = strtotime( $content, $date );
 
-	// if strtotime doesn't work, try stronger measures
-	if ( $time === false || $time === -1 ) {
-		$timearray = date_parse( $content );
-		foreach ( $timearray as $key => $value ) {
-			if ( $value === false ) unset ( $timearray[$key] );
+	/**
+	 * Process the time shortcode
+	 *
+	 * @param array $attr (unused)
+	 * @param string $content - Any valid strtotime string
+	 * @return string - html content
+	 **/
+	public static function time_shortcode( $attr, $content = null ) {
+
+		global $in_comment_content, $comment;
+
+		// try to parse the time, relative to the post/comment time
+		if ( $in_comment_content ) {
+			$relative_time = strtotime( $comment->comment_date_gmt );
+		} else {
+			$relative_time = get_the_date( 'U' );
 		}
 
-		// merge the info from the post date with this one
-		$relative = date_parse( $raw_date );
-		$timearray = array_merge( $relative, $timearray );
+		$time = self::parse_time( $content, $relative_time );
 
-		// use the blog's timezone if none was specified
-		if ( !isset( $timearray['tz_id'] ) ) {
-			$timearray['tz_id'] = get_option( 'timezone_string' );
+		// if that didn't work, give up
+		if ( $time === false || $time === -1 ) {
+			return $content;
 		}
 
-		// build a normalized time string, then parse it to an integer time using strtotime
-		$time = strtotime( "{$timearray['year']}-{$timearray['month']}-{$timearray['day']}T{$timearray['hour']}:{$timearray['minute']}:{$timearray['second']} {$timearray['tz_id']}" );
+		// build the link and abbr microformat
+		$out = '<a href="http://www.timeanddate.com/worldclock/fixedtime.html?iso=' . gmdate( 'Ymd\THi', $time ) . '"><abbr class="globalized-date" title="' . gmdate( 'c', $time ) . '" data-time="' . absint( $time ) . '000">' . $content . '</abbr></a>';
+
+		// add the time converter JS code if not already added
+		if( !has_action( 'wp_footer', array( 'o2_Time_Shortcode', 'time_conversion_script' ) ) ){
+			add_action( 'wp_footer', array( 'o2_Time_Shortcode', 'time_conversion_script' ) );
+		}
+
+		// return the new link
+		return $out;
 	}
 
-	// if that didn't work, give up
-	if ( $time === false || $time === -1 ) {
-		return $content;
+	/**
+	 * Convert a time string into an epoch timestamp
+	 *
+	 * @param string $date_string - Any valid strtotime string
+	 * @param int $time (optional) - Epoch timestamp for relative date parsing
+	 * @return mixed - int Epoch timestamp on success, false on fail
+	 **/
+	public static function parse_time( $date_string, $time = null ) {
+
+		if( empty( $date_string ) ){
+			return false;
+		}
+
+		if( empty( $time ) ){
+			$time = time();
+		}
+
+		$html_entity_regex = '/&[^\s]*;/';
+		$date_string = preg_replace( $html_entity_regex, ' ', $date_string );
+		$date_string = str_ireplace( "U+00A0", ' ', $date_string );
```

Comment:
There's no need to handle the string "U+00A0". The original problem was with a no-break space: Unicode codepoint U+00A0. In UTF-8, that's the byte sequence "\xC2\xA0".